### PR TITLE
Change loss type names

### PR DESCRIPTION
**Pull request by ahwillia**
Changes all loss function types to have CamelCase names, preventing naming conflicts with StatsBase. See: https://github.com/madeleineudell/LowRankModels.jl/issues/41

I can also rename all the regularization types to CamelCase if desired...

### CODE REVIEW

**Comment by madeleineudell** [comment]
This is great! Yes, let's go ahead and change the regularizers, too, for consistency.

Would you mind deprecating the old losses and adding them to utilities/deprecated.jl? That way we'll be able to give some notice (via warnings instead of errors) to users who don't read the documentation every morning.

**Comment by ahwillia** [comment]
Nice. I incorporated these changes. What is your opinion of the new names such as:

`nonnegative()` => `NonnegConstraint()`
`unitonesparse()` => `UnitOneSparseConstraint()`
`simplex()` => `SimplexConstraint()`
etc.

They are a little more cumbersome to type but are more explicit, which I think might be a good tradeoff in this case

File: README.md [hunk 1]
```diff
@@ -69,24 +69,25 @@ and may be chosen from a list of supported losses and regularizers, which includ
...
-* quadratic regularization `quadreg`
-* constrained squared euclidean norm `constrained_quadreg`
-* l1 regularization `onereg`
-* no regularization `zeroreg`
-* nonnegative constraint `nonnegative` (eg, for nonnegative matrix factorization)
-* 1-sparse constraint `onesparse` (eg, for orthogonal NNMF)
-* unit 1-sparse constraint `unitonesparse` (eg, for k-means)
-* simplex constraint `simplex`
-* l1 regularization, combined with nonnegative constraint `nonneg_onereg`
+* QuadLoss regularization `QuadReg`
```
Comment (madeleineudell) [inline README.md]: let's stick with `quadratic` regularization, not `QuadLoss`. somehow the latter sounds like we're trying to copyright least squares.

File: README.md [hunk 2]
```diff
@@ -69,24 +69,25 @@ and may be chosen from a list of supported losses and regularizers, which includ
...
+* constrained squared euclidean norm `ConstrainedQuadReg`
```
Comment (madeleineudell) [inline README.md]: `QuadConstraint` ?

File: README.md [hunk 3]
```diff
@@ -69,24 +69,25 @@ and may be chosen from a list of supported losses and regularizers, which includ
...
+* l1 regularization `OneReg`
+* no regularization `ZeroReg`
+* nonnegative constraint `NonnegConstraint` (eg, for nonnegative matrix factorization)
```
Comment (madeleineudell) [inline README.md]: capitalize Neg: NonNegConstraint

File: README.md [hunk 4]
```diff
@@ -69,24 +69,25 @@ and may be chosen from a list of supported losses and regularizers, which includ
...
+* 1-sparse constraint `OneSparseConstraint` (eg, for orthogonal NNMF)
+* unit 1-sparse constraint `UnitOneSparseConstraint` (eg, for k-means)
+* simplex constraint `SimplexConstraint`
+* l1 regularization, combined with nonnegative constraint `NonnegOneReg`
```
Comment (madeleineudell) [inline README.md]: ditto: NonNegOneReg

**Comment by madeleineudell** [comment]
Yes, I like distinguishing the ones that impose a hard constraint. Minor
line comments above; other than that looks good to merge (once we've run
tests, etc).

On Wed, Sep 16, 2015 at 12:12 PM, Alex Williams <notifications@github.com>
wrote:

> Nice. I incorporated these changes. What is your opinion of the new names
> such as:
>
> nonnegative() => NonnegConstraint()
> unitonesparse() => UnitOneSparseConstraint()
> simplex() => SimplexConstraint()
> etc.
>
> They are a little more cumbersome to type but are more explicit, which I
> think might be a good tradeoff in this case
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/madeleineudell/LowRankModels.jl/pull/42#issuecomment-140850242>
> .
>



-- 
Madeleine Udell
Postdoctoral Fellow at the Center for the Mathematics of Information
California Institute of Technology
www.stanford.edu/~udell
(415) 729-4115

**Comment by ahwillia** [comment]
Tried to consolidate the last two commits together to tidy things up. Tests all pass for me.

**Comment by madeleineudell** [comment]
Perfect! Merging.

**Pull request merged**